community/kernel-api/src/main/java/org/neo4j/internal/kernel/api/SchemaRead.java

```diff
@@ -32,4 +37,31 @@
      * @return the IndexReference, or {@link CapableIndexReference#NO_INDEX} if such an index does not exist.
      */
     CapableIndexReference index( int label, int... properties );
+
+    /**
+     * Finds all constraints for the given schema
+     * @param descriptor The descriptor of the schema
+     * @return All constraints for the given schema
+     */
+    Iterator<ConstraintDescriptor> constraintsGetForSchema( SchemaDescriptor descriptor );
```

Comment:
My understanding has been that we intend to use `IndexReference` as key for the index mapping and leave `SchemaDescriptor` as an internal thing. That's the reason why `index( int label, int... properties )` return `CapableIndexReference` and why `interface Read` use `IndexReference` everywhere.

Are we abandoning this plan? If so, `IndexReference` should be cleaned up. If not, we should be consistent about on what layer we switch from `IndexReference` to `IndexDescriptor`.

What are your thoughts about this?

Reply:
I wasn't aware of that plan. I am actually terribly confused over the distinction between `IndexDescriptor` versus `IndexReference`. This API is actually not needed for this PR we could have managed with just internal API. Could we fix this API later in a separate PR?

Reply:
Yeah, I'm confused as well 😅  Let's clean this up in separate PR. I would be happy to help!

Reply:
Thanks, we should maybe have a `ConstraintReference` as well

Reply:
At least!